clients/unshaded/src/main/java/tachyon/client/OutStreamBase.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import tachyon.annotation.PublicApi;
+
+/**
+ * An abstraction of the output stream API in Tachyon to write data to a file or a block. In
+ * addition to extending abstract class {@link OutputStream} as the basics, it also keeps counting
+ * the number of bytes written to the output stream, and extends {@link Cancelable} to abort the
+ * writes.
+ */
+@PublicApi
+public abstract class OutStreamBase extends OutputStream implements Cancelable {
+  // TODO(binfan): make mBytesWritten long.
```

Comment:
in the codebase, there are some places assuming the offset is okey to be as an int (i.e., offset is no more than 2GB). So right now using int to denote the offset is fine. This todo is really about removing this assumption in the codebase together.